code/datums/components/bump_attack.dm

```diff
@@ -69,33 +69,47 @@
 
 /datum/component/bump_attack/proc/human_bump_action(datum/source, atom/target)
 	SIGNAL_HANDLER
-	var/mob/living/carbon/human/bumper = parent
+	INVOKE_ASYNC(src, .proc/human_bump_action_checks, source, target)
+
+/datum/component/bump_attack/proc/human_bump_action_checks(datum/source, atom/target)
+	var/mob/living/carbon/human/attacker = bumper
 	. = carbon_bump_action_checks(target)
 	if(!isnull(.))
 		return
 	var/mob/living/living_target = target
-	if(bumper.faction == living_target.faction)
+	if(attacker.faction == living_target.faction)
 		return //FF
```

Comment:
??

Reply:
removed var, INVOKE_ASYNC is needed for sleeping procs